### ISSUE TITLE
Chart freezes with two series, one empty

### ISSUE THREAD

**Issue by wezm** [amcharts/amcharts4#300]
Hi, I've run into an issue whilst evaluating amCharts for our product. I have two series with data set on the series directly. One series has data, the other is empty. After loading the chart it freezes if you hover over it. In Chrome the process just hangs and needs to be killed. In Firefox an alert is shown saying that a script is slowing the page down and you can click a button to stop it.

From my investigations the chart freezes because it enters an infinite loop in `DataAxis.getSeriesDataItem`. Specifically in this `while` loop: https://github.com/amcharts/amcharts4/blob/b7a274c6ecba34971cbcfd3ec564d92d78cdf9f7/dist/es2015/.internal/charts/axes/DateAxis.js#L1343

This Pen reproduces the issue: https://codepen.io/wezm/pen/NLRrgJ/?editors=0010

The source from the pen is:

```javascript
am4core.useTheme(am4themes_animated);

var chart = am4core.create("chartdiv", am4charts.XYChart);

var data1 = [];
var data2 = [];

var price1 = 1000, price2 = 1200;
var quantity = 30000;

data1 = [{"date":1535433426000,"value":-9005},{"date":1535433600000,"value":-9005},{"date":1535433900000,"value":-3952},{"date":1535434200000,"value":0},{"date":1535434500000,"value":-47589},{"date":1535434800000,"value":-42313}];

var dateAxis = chart.xAxes.push(new am4charts.DateAxis());
dateAxis.renderer.grid.template.location = 0;
dateAxis.renderer.labels.template.fill = am4core.color("#e59165")

var valueAxis = chart.yAxes.push(new am4charts.ValueAxis());
valueAxis.tooltip.disabled = false;
valueAxis.renderer.labels.template.fill = am4core.color("#e59165");

valueAxis.renderer.minWidth = 60;

var series = chart.series.push(new am4charts.StepLineSeries());
series.name = "2016";
series.dataFields.dateX = "date";
series.dataFields.valueY = "value";
series.tooltipText = "{valueY.value}";
series.fill = am4core.color("#e59165");
series.stroke = am4core.color("#e59165");
//series.strokeWidth = 3;
series.data = data1;

var series2 = chart.series.push(new am4charts.StepLineSeries());
series2.name = "2017";
series2.dataFields.dateX = "date2";
series2.dataFields.valueY = "price2";
// series2.yAxis = valueAxis2;
// series2.xAxis = dateAxis2;
series2.tooltipText = "{valueY.value}";
series2.fill = am4core.color("#dfcc64");
series2.stroke = am4core.color("#dfcc64");
//series2.strokeWidth = 3;
series2.data = data2;

chart.cursor = new am4charts.XYCursor();
chart.cursor.xAxis = dateAxis2;

var scrollbarX = new am4charts.XYChartScrollbar();
scrollbarX.series.push(series);
chart.scrollbarX = scrollbarX;

chart.legend = new am4charts.Legend();
chart.legend.parent = chart.plotContainer;
chart.legend.zIndex = 100;

valueAxis2.renderer.grid.template.strokeOpacity = 0.07;
dateAxis2.renderer.grid.template.strokeOpacity = 0.07;
dateAxis.renderer.grid.template.strokeOpacity = 0.07;
valueAxis.renderer.grid.template.strokeOpacity = 0.07;
```

**Comment by martynasma** [comment]
Looks like a bug. Thanks for the heads up!

Will be fixing it. I expect a fix early next week.

**Comment by martynasma** [comment]
We've been able to distill this issue.

It's posted as a separate issue here: #301 

Make sure you subscribe to #301 to keep yourself updated about the progress.

Thanks!

**Comment by martynasma** [comment]
We've been able to fix this and will release an update soon. (within couple of days)

**Comment by wezm** [comment]
Great, thanks for the quick fix :+1:

**Comment by martynasma** [comment]
Fixed in [4.0.0-beta.48](https://github.com/amcharts/amcharts4/releases/tag/4.0.0-beta.48).

## [4.0.0-beta.48] - 2018-08-31

### Added
- `useWebFonts` setting in `Export`. Allows disabling of download of web fonts when exporting.
- `useRetina` setting in `Export`. Images are now exported scaled up on retina displays by default. Set `useRetina = false` to disable.

### Changed
- Export fallback mechanism for older browsers changed from FabricJS to canvg, which is much lighter and has better text formatting support.
- On retina displays images will now export supersized in order not to lose quality. Set `useRetina = false` to disable.
- amCharts logo will now auto-hide on supersmall chart sizes.

### Fixed
- Cursor was zooming the chart even if drag motion was performed on an external element positioned over the chart.
- Cursor was not hidden when it was no longer hovering the chart area if chart had no padding.
- `urlTarget` was not carried over from `ListTemplate` template to new items.
- Cursor could cause chart to freeze up on a DateAxis with empty cells in the axis' beginning or end.
- Export will now correctly handle web fonts, such as Google Fonts. Please note that FontAwesome is not supported, since they forbid any kind of usage, except inclusion via CSS.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
